app/ui/src/main/java/com/fsck/k9/activity/ChooseFolder.java

```diff
@@ -224,6 +224,8 @@ private void onRefresh() {
     }
 
     private void setDisplayMode(FolderMode aMode) {
+        mAccount.setFolderTargetMode(aMode);
+        Preferences.getPreferences(getApplicationContext()).saveAccount(mAccount);
```

Comment:
I have not found a `saveAccount()` call that is not in the UI thread. Do you have an example for some code that is executed on an IO thread? Then I can make it consistent with other places.

> I'm not sure if fixing this is worth it. I want us to get rid of folder classes.

Do you plan to do all those refactorings before releasing the next version? When reading the comments on Google Play, there are quite a few users who assume that the project is dead because it was a long time without an update. It might be better to release more often and change things progressively, rather than working on a "big bang" release for years. Google also suggests to [publish updates regularly](https://support.google.com/googleplay/android-developer/answer/4448378?hl=en). I don't want to sound like I'm complaining (after all, I can do a fresh build as often as I want) but getting the redesign out rather sooner than later might prevent users from going back to apps like Gmail.

Is the [5.800 milestone](https://github.com/k9mail/k-9/milestone/24) up-to-date with things you consider blockers for a new release?

Reply:
> I have not found a saveAccount() call that is not in the UI thread. Do you have an example for some code that is executed on an IO thread?

One example can be found in `AccountSettingsDataStore`: https://github.com/k9mail/k-9/blob/1c29f1c09ad1f60105df9055c5481c8839de0eeb/app/ui/src/main/java/com/fsck/k9/ui/settings/account/AccountSettingsDataStore.kt#L207-L215

> It might be better to release more often [...]

I agree. I'm not saying we need to fix all known issues. What I was trying to say was that we shouldn't introduce code now that we know needs to be fixed later.
Given that the behavior has been like this for ages, I see no reason to change it now. It might even be like this on purpose (override the setting this once, don't permanently change it; there's another place to do the permanent change 'Account settings > Folders > Move/copy destination folders').

> Is the 5.800 milestone up-to-date with things you consider blockers for a new release?

Mostly. Although I'm guessing some issues will join this list as we get closer to a release.